eudemo/eudemo/maintenance/lower_port.py

```diff
@@ -0,0 +1,122 @@
+
+from bluemira.base.designer import Designer
+from bluemira.base.parameter_frame import Parameter, ParameterFrame
+from bluemira.geometry.coordinates import Coordinates
+from bluemira.geometry.face import BluemiraFace
+from bluemira.geometry.tools import make_polygon
+from bluemira.geometry.wire import BluemiraWire
+
+
+@dataclass
+class LowerPortDesignerParams(ParameterFrame):
+    """LowerPort ParameterFrame"""
+
+    lower_port_angle: Parameter[float]
+    divertor_padding: Parameter[float]
+
+
+class LowerPortDesigner(Designer):
+    """
+    Lower Port Designer
+
+    Notes
+    -----
+    Retrictions on angle is ±45° because there are no protections
+    keeping the port dimensions constant
+
+    """
+
+    param_cls: Type[ParameterFrame] = LowerPortDesignerParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame],
+        build_config: Dict,
+        divertor_xz: BluemiraFace,
+        x_wall_inner: float,
+        x_wall_outer: float,
+        x_extrema: float,
+    ):
+        super().__init__(params, build_config)
+        self.divertor_xz = divertor_xz
+        self.x_wall_inner = x_wall_inner
+        self.x_wall_outer = x_wall_outer
+        self.x_extrema = x_extrema
+
+    def run(self) -> Tuple[BluemiraFace, BluemiraWire]:
+        """Run method of Designer"""
+        if not -45 <= self.params.lower_port_angle.value <= 45:
+            raise ValueError(
+                f"{self.params.lower_port_angle.name}"
+                " must be within the range -45 < x < 45 degrees:"
+                f" {self.params.lower_port_angle.value}"
+            )
+
+        z_start = self.divertor_xz.bounding_box.z_min
+        z_outer = z_start + (self.x_wall_inner - self.x_wall_outer) * np.tan(
+            np.deg2rad(self.params.lower_port_angle.value)
+        )
+        x_start = self.divertor_xz.center_of_mass[0]
+
+        x_path = np.array(
+            [x_start, self.x_wall_inner, self.x_wall_outer, self.x_extrema]
+        )
+        z_lower_path = np.array([z_start, z_start, z_outer, z_outer])
+
+        traj = make_polygon(
+            Coordinates(
+                {
+                    "x": x_path,
+                    "z": z_lower_path - self.params.divertor_padding.value,
+                }
+            )
+        )
+
+        port = make_polygon(
+            Coordinates(
+                {
+                    "x": np.append(x_path, x_path[::-1]),
+                    "z": np.append(
+                        z_lower_path - self.params.divertor_padding.value,
+                        (
+                            z_lower_path
+                            + (
+                                self.divertor_xz.bounding_box.z_max
+                                + self.params.divertor_padding.value,
+                            )
+                        )[::-1],
+                    ),
+                }
+            ),
+            closed=True,
+        )
+
+        return BluemiraFace(port), traj
```

Comment:
Why are both the trajectory and the exclusion zone well below the divertor? Surely the exclusion zone should encompass the divertor, no?